paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -396,6 +418,79 @@ def sync_secrets(
     return True
 
 
+def sync_database_credentials(
+    kube_client: KubeClient,
+    cluster: str,
+    service: str,
+    secret_provider_name: str,
+    vault_cluster_config: Dict[str, str],
+    soa_dir: str,
+    vault_token_file: str,
+) -> bool:
+    config_loader = PaastaServiceConfigLoader(service=service, soa_dir=soa_dir)
+    for instance_config in config_loader.instance_configs(
+        cluster=cluster, instance_type_class=KubernetesDeploymentConfig
+    ):
+        # expects VAULT_ADDR_OVERRIDE, VAULT_CA_OVERRIDE, and VAULT_TOKEN_OVERRIDE to be set
```

Comment:
++ found a possible implementation: https://stackoverflow.com/questions/2059482/temporarily-modify-the-current-processs-environment. were you thinking we add this in as hardcoded values for now, or add them as keyword arguments to the paasta_secrets_sync script?

alternatively, I was thinking we do it this way. let me know if I'm mistaken:

```
# https://sourcegraph.yelpcorp.com/sysgit/puppet/-/blob/modules/profile_kube/manifests/management.pp?L169:178
command                => 'bash -o pipefail -c "[...] | xargs VAULT_ADDR_OVERRIDE=<> VAULT_CA_OVERRIDE=<> VAULT_TOKEN_OVERRIDE=<> paasta_secrets_sync --secret_type database-credentials; fi"', 
```

then the variable is only set during that single bash run

running it manually would require us to add `--secret_type database-credentials` as well since this vault shard has different data from the main vault shard for boto/crypto keys